### PR TITLE
DOC: Add example code snippets to `jax.scipy.linalg.expm` and `jax.scipy.linalg.polar` docs

### PR DESCRIPTION
**Pull request by rajasekharporeddy**

### CODE REVIEW

File: jax/_src/scipy/linalg.py [hunk 1]
```diff
@@ -1137,6 +1137,21 @@ def expm(A: ArrayLike, *, upper_triangular: bool = False, max_squarings: int = 1
 
   See Also:
     :func:`jax.scipy.linalg.expm_frechet`
+
+  Example:
```
Comment (jakevdp) [inline jax/_src/scipy/linalg.py:1141]: I think this is a poor example. Examples should show give insight into why a function is useful, not simply show that you can call a function and get numbers out without any broader context.

What if we demonstrated one of the properties of the matrix exponential; e.g. that `expm(A + B)` is equivalent to `expm(A) @ expm(B)` for commuting matrices `A` and `B`?

File: jax/_src/scipy/linalg.py [hunk 2]
```diff
@@ -1642,6 +1657,37 @@ def polar(a: ArrayLike, side: str = 'right', *, method: str = 'qdwh', eps: float
...
+
+    >>> a = jnp.array([[1., 2., 3.],
+    ...                [5., 4., 2.],
+    ...                [3., 2., 1.]])
+    >>> U, P = jax.scipy.linalg.polar(a)
+
+    U is a Unitary Matrix:
+
+    >>> with jnp.printoptions(precision=2, suppress=True):
+    ...     print(U)
```
Comment (jakevdp) [inline jax/_src/scipy/linalg.py:1672]: Maybe print `U.T @ U` to show that the result is the identity?

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/scipy/linalg.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/scipy/linalg.py:1141]: Thanks for the suggestion. Updated accordingly.

File: jax/_src/scipy/linalg.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/scipy/linalg.py]: Thanks for the suggestion. Updated accordingly.

File: jax/_src/scipy/linalg.py [hunk 3]
```diff
@@ -1137,6 +1137,46 @@ def expm(A: ArrayLike, *, upper_triangular: bool = False, max_squarings: int = 1
...
+    >>> B = jnp.array([[3, 0],
+    ...                [0, 4]])
+    >>> with jnp.printoptions(precision=1, suppress=True):
+    ...     print(jax.scipy.linalg.expm(A+B))
+    [[148.4   0. ]
+     [  0.  148.4]]
+    >>> with jnp.printoptions(precision=1, suppress=True):
+    ...     print(jax.scipy.linalg.expm(A) @ jax.scipy.linalg.expm(B))
+    [[148.4   0. ]
+     [  0.  148.4]]
```
Comment (jakevdp) [inline jax/_src/scipy/linalg.py:1157]: You could even avoid printing these and just do the allclose

File: jax/_src/scipy/linalg.py [hunk 4]
```diff
@@ -1137,6 +1137,46 @@ def expm(A: ArrayLike, *, upper_triangular: bool = False, max_squarings: int = 1
 
   See Also:
     :func:`jax.scipy.linalg.expm_frechet`
+
+  Example:
+
+    If the matrices ``A`` and ``B`` satisfies ``A @ B = B @ A``, then
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/scipy/linalg.py]: Yes adding more context is a good idea. Thanks @jakevdp. Updated the context before the example.

File: jax/_src/scipy/linalg.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/scipy/linalg.py]: Done

File: jax/_src/scipy/linalg.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/scipy/linalg.py]: Looks great, thanks!

**Comment by rajasekharporeddy** [comment]
Hi @jakevdp, Could you please have a look at this. All changes are made as suggested and squashed into single commit.

**Review by jakevdp — APPROVED** [review]
Looks good!

**Pull request merged**